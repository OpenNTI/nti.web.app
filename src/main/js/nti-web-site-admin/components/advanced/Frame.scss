.advanced-admin-tools-view {
	margin-top: 40px;
}
