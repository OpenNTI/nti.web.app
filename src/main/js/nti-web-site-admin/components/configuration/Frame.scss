.advanced-admin-tools-view {
	margin-top: 20px;
}
