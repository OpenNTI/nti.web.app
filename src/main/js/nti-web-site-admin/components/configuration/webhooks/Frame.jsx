import { scoped } from '@nti/lib-locale';
import { DataContext } from '@nti/web-core/data';
import { LinkTo } from '@nti/web-routing';
import { Box, Icons, Text } from '@nti/web-core';

import { Store as Store } from './Store';

const t = scoped('nti-web-app.admin.config.Webhooks', {
	title: 'Webhook Subscriptions',
	back: 'Back to Subscriptions',
});

const Header = styled.div`
	padding: 1rem 0;

	& a {
		color: var(--primary-grey);
		display: flex;
		flex-direction: row;
		align-items: center;
	}

	& i {
		font-size: 0.875rem;
		font-weight: 700;
		width: 1rem;
	}
`;

export const Frame = ({ children, ...props }) => {
	return (
		<Box p="lg" sh="sm">
			{children && (
				<Header>
					<LinkTo.Name name="site-admins.config.webhooks">
						<Icons.Chevron.Left />
						<span>{t('back')}</span>
					</LinkTo.Name>
				</Header>
			)}
			<Text as="h1">{t('title')}</Text>
			<DataContext store={Store.useStore()} fallback={<div />}>
				{children}
			</DataContext>
		</Box>
	);
};
