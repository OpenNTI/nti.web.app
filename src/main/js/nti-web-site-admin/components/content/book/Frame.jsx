import React from 'react';
import PropTypes from 'prop-types';
import {LinkTo} from '@nti/web-routing';
import {Loading, Layouts} from '@nti/web-commons';
import {scoped} from '@nti/lib-locale';
import {decodeFromURI} from '@nti/lib-ntiids';

import Store from '../info/Store';

import NavBar from './nav-bar';

const DEFAULT_TEXT = {
	back: 'Back to Books'
};
const t = scoped('nti-site-admin.courses.book.Frame', DEFAULT_TEXT);

export default
@Store.connect({course: 'course', loading: 'loading'})
class SiteAdminBookView extends React.Component {
	static deriveStoreKeyFromProps (props) {
		return props.bookID;
	}

	static propTypes = {
		bookID: PropTypes.string.isRequired,

		course: PropTypes.object,
		loading: PropTypes.bool,
		store: PropTypes.object,

		children: PropTypes.element
	}

	get store () {
		return this.props.store;
	}


	getCourseID (props = this.props) {
		const {bookID} = props;

		return decodeFromURI(bookID);
	}


	componentDidMount () {
		this.store.loadCourse(this.getCourseID());
	}


	componentDidUpdate (prevProps) {
		const newID = this.getCourseID(this.props);
		const oldID = this.getCourseID(prevProps);

		if (newID !== oldID) {
			this.store.loadUser(newID);
		}
	}


	render () {
		const {loading} = this.props;

		return (
			<div className="site-admin-course-view">
				{loading && this.renderHeader()}
				{loading && (<Loading.Mask />)}
				{!loading && this.renderCourse()}
			</div>
		);
	}


	renderCourse () {
		const {course, children} = this.props;

		if (!course) {
			return null;
		}

		return (
			<Layouts.NavContent.Container>
				<Layouts.NavContent.Nav className="nav-bar">
					{this.renderHeader()}
					<NavBar book={course} />
				</Layouts.NavContent.Nav>
				<Layouts.NavContent.Content className="content">
					{React.Children.map(children, (item) => {
						return React.cloneElement(item, {course});
					})}
				</Layouts.NavContent.Content>
			</Layouts.NavContent.Container>
		);
	}

	renderHeader () {
		return (
			<div className="header">
				<LinkTo.Name name="site-admin.content.content-list-books">
					<i className="icon-chevron-left" />
					<span>{t('back')}</span>
				</LinkTo.Name>
			</div>
		);
	}
}
