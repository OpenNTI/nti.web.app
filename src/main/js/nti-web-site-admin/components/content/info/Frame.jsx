import React from 'react';
import PropTypes from 'prop-types';
import {LinkTo} from '@nti/web-routing';
import {Loading, Layouts} from '@nti/web-commons';
import {scoped} from '@nti/lib-locale';
import {decodeFromURI} from '@nti/lib-ntiids';

import {getString} from 'legacy/util/Localization';

import Store from './Store';
import NavBar from './nav-bar';

const DEFAULT_TEXT = {
	back: getString('NextThought.view.Navigation.back')
};
const t = scoped('nti-site-admin.courses.info.Frame', DEFAULT_TEXT);

@Store.connect({course: 'course', loading: 'loading'})
export default class SiteAdminCourseView extends React.Component {
	static propTypes = {
		courseID: PropTypes.string.isRequired,

		course: PropTypes.object,
		loading: PropTypes.bool,
		store: PropTypes.object,

		children: PropTypes.element
	}

	get store () {
		return this.props.store;
	}


	getCourseID (props = this.props) {
		const {courseID} = props;

		return decodeFromURI(courseID);
	}


	componentDidMount () {
		this.store.loadCourse(this.getCourseID());
	}


	componentDidUpdate (prevProps) {
		const newID = this.getCourseID(this.props);
		const oldID = this.getCourseID(prevProps);

		if (newID !== oldID) {
			this.store.loadUser(newID);
		}
	}


	render () {
		const {loading} = this.props;

		return (
			<div className="site-admin-course-view">
				{loading && this.renderHeader()}
				{loading && (<Loading.Mask />)}
				{!loading && this.renderCourse()}
			</div>
		);
	}


	renderCourse () {
		const {course, children} = this.props;

		if (!course) {
			return null;
		}

		return (
			<Layouts.NavContent.Container>
				<Layouts.NavContent.Nav className="nav-bar">
					{this.renderHeader()}
					<NavBar course={course} />
				</Layouts.NavContent.Nav>
				<Layouts.NavContent.Content className="content">
					{React.Children.map(children, (item) => {
						return React.cloneElement(item, {course});
					})}
				</Layouts.NavContent.Content>
			</Layouts.NavContent.Container>
		);
	}

	renderHeader () {
		return (
			<div className="header">
				<LinkTo.Name name="site-admin.courses">
					<i className="icon-chevron-left" />
					<span>{t('back')}</span>
				</LinkTo.Name>
			</div>
		);
	}
}
