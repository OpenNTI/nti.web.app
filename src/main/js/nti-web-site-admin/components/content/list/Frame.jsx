import React from 'react';

import { Layouts, FixedElement } from '@nti/web-commons';

import NavBar from './nav-bar';

export default function SiteAdminContentListFrame({ children }) {
	return (
		<div
			className="site-admin-content-list-frame"
			css={css`
				margin-top: 20px;
			`}
		>
			<Layouts.NavContent.Container>
				<Layouts.NavContent.Nav className="nav-bar">
					{/* {this.renderHeader()} */}
					<FixedElement>
						<NavBar />
					</FixedElement>
				</Layouts.NavContent.Nav>
				<Layouts.NavContent.Content className="content">
					{React.Children.map(children, item => {
						return React.cloneElement(item);
					})}
				</Layouts.NavContent.Content>
			</Layouts.NavContent.Container>
		</div>
	);
}
