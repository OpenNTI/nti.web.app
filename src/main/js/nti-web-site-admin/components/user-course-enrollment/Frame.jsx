import React from 'react';
import PropTypes from 'prop-types';
import {LinkTo} from 'nti-web-routing';// eslint-disable-line
import {Loading, Layouts} from 'nti-web-commons';
import {scoped} from 'nti-lib-locale';

import Store from './Store';
import NavBar from './nav-bar';

const DEFAULT_TEXT = {
	back: 'Back to User'
};

const t = scoped('nti-site-admin.users.user.user-course-enrollment.Frame', DEFAULT_TEXT);

@Store.connect({enrollment: 'enrollment', loading: 'loading'})
export default class SiteAdminUserEnrollmentView extends React.Component {
	static propTypes = {
		enrollmentID: PropTypes.string,
		loading: PropTypes.bool,
		store: PropTypes.object,

		enrollment: PropTypes.object,

		children: PropTypes.node
	}

	get store () {
		return this.props.store;
	}


	componentWillReceiveProps (nextProps) {
		const {enrollmentID: newID} = nextProps;
		const {enrollmentID: oldID} = this.props;

		if (newID !== oldID) {
			this.store.loadEnrollment(newID);
		}
	}


	componentDidMount () {
		const {enrollmentID} = this.props;

		this.store.loadEnrollment(enrollmentID);
	}


	render () {
		const {loading} = this.props;

		return (
			<div className="site-admin-user-enrollment-view">
				{loading && this.renderHeader()}
				{loading && (<Loading.Mask />)}
				{!loading && this.renderUser()}
			</div>
		);

	}

	renderUser () {
		const {children, enrollment} = this.props;

		return (
			<Layouts.NavContent.Container>
				<Layouts.NavContent.Nav className="nav-bar">
					{this.renderHeader()}
					<NavBar enrollment={enrollment}/>
				</Layouts.NavContent.Nav>
				<Layouts.NavContent.Content className="content">
					{React.Children.map(children, (item) => {
						return React.cloneElement(item, {enrollment});
					})}
				</Layouts.NavContent.Content>
			</Layouts.NavContent.Container>
		);
	}


	renderHeader () {
		return (
			<div className="header">
				<LinkTo.Name name="site-admin.users.user-overview">
					<i className="icon-chevron-left" />
					<span>{t('back')}</span>
				</LinkTo.Name>
			</div>
		);
	}
}
