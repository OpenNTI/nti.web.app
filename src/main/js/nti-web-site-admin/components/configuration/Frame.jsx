import './Frame.scss';
import React from 'react';
import PropTypes from 'prop-types';
import {LinkTo} from '@nti/web-routing';// eslint-disable-line
import {Layouts, FixedElement} from '@nti/web-commons';


import NavBar from './nav-bar';

export default class SiteAdminAdvancedView extends React.Component {
	static propTypes = {
		children: PropTypes.node
	}

	render () {
		return (
			<div className="advanced-admin-tools-view">
				{this.renderLayout()}
			</div>
		);

	}

	renderLayout () {
		const {children} = this.props;

		return (
			<Layouts.NavContent.Container>
				<Layouts.NavContent.Nav className="nav-bar">
					<FixedElement>
						<NavBar/>
					</FixedElement>
				</Layouts.NavContent.Nav>
				<Layouts.NavContent.Content className="content">
					{React.Children.map(children, (item) => {
						return React.cloneElement(item, {});
					})}
				</Layouts.NavContent.Content>
			</Layouts.NavContent.Container>
		);
	}
}
