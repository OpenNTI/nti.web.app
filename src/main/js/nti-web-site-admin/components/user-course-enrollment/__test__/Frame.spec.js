import React from 'react';
import renderer from 'react-test-renderer';
import { TestUtils } from '@nti/web-client';

import Frame from '../Frame';

const { tearDownTestClient, setupTestClient } = TestUtils;

const getMockService = () => {
	return {
		getObject: async () => {
			return {
				fetchLinkParsed: () => {
					return {
						course: {
							CatalogEntry: {
								Title: 'course1',
								getDefaultAssetRoot () { return 'testRoot'; }
							}
						}
					};
				},
				UserProfile: 'testUser'
			};
		}
	};
};

const onBefore = () => {
	jest.useFakeTimers();
	setupTestClient(getMockService());
};

const onAfter = () => {
	tearDownTestClient();
};

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

/* eslint-env jest */
describe('Site admin user course enrollment frame', () => {
	beforeEach(() => onBefore());
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<Frame/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
