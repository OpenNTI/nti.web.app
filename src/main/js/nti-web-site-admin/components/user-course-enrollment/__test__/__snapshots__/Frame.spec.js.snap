// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user course enrollment frame Basic render test 1`] = `
<div
  className="site-admin-user-enrollment-view"
>
  <div
    className="nav-content-container"
  >
    <div
      className="nav-content-nav nav-bar"
    >
      <div
        className="header"
      >
        <div
          className="nti-link-to-path"
          to="site-admin.users.user-overview"
        >
          <i
            className="icon-chevron-left"
          />
          <span>
            Back to User
          </span>
        </div>
      </div>
      <div
        className="site-admin-card site-admin-user-enrollment-nav-bar"
      >
        <div
          className="site-admin-user-enrollment-identity"
        >
          <div
            className="profile-image"
          >
            <img
              className="enrollment-course-image"
              src=""
            />
            <div
              className="user-avatar"
            >
              <img
                alt="Avatar for Anonymous User"
                className="avatar avatar-color-6"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              />
            </div>
          </div>
          <div
            className="enrollment-info"
          >
            <div
              className="plain-text"
            >
              Activity for
            </div>
            <span
              className="username"
              rel="author"
            >
              Anonymous User
            </span>
            <div
              className="plain-text"
            >
              in
            </div>
            <div
              className="course-title"
            />
          </div>
        </div>
        <ul
          className="unadorned-list site-admin-user-tabs"
        >
          <li>
            <div
              className="nti-link-to-path"
              exact={true}
              to="./"
            >
              Overview
            </div>
          </li>
          <li>
            <div
              className="nti-link-to-path"
              to="./reports"
            >
              Reports
            </div>
          </li>
          <li>
            <div
              className="nti-link-to-path"
              to="./progress"
            >
              Progress
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="nav-content-content content"
    />
  </div>
</div>
`;
